Failed to compile on ubuntu 20 with golang 1.17.7

Failed to compile manually

/opt/apps/reactivesearch-api$ make
go build --buildmode=plugin -o build/plugins/auth.so plugins/auth/main/auth.go
# github.com/cespare/xxhash
asm: xxhash_amd64.s:120: when dynamic linking, R15 is clobbered by a global variable access and is used here: 00092 (/home/tomburnell/go/pkg/mod/github.com/cespare/xxhash@v1.1.0/xxhash_amd64.s:120)	ADDQ	R15, AX
asm: assembly failed
# github.com/cespare/xxhash/v2
asm: xxhash_amd64.s:120: when dynamic linking, R15 is clobbered by a global variable access and is used here: 00092 (/home/tomburnell/go/pkg/mod/github.com/cespare/xxhash/v2@v2.1.1/xxhash_amd64.s:120)	ADDQ	R15, AX
asm: assembly failed
# github.com/klauspost/compress/zstd/internal/xxhash
asm: xxhash_amd64.s:120: when dynamic linking, R15 is clobbered by a global variable access and is used here: 00092 (/home/tomburnell/go/pkg/mod/github.com/klauspost/compress@v1.12.3/zstd/internal/xxhash/xxhash_amd64.s:120)	ADDQ	R15, AX
asm: assembly failed
make: *** [Makefile:20: plugins/auth/main/auth.so] Error 2

### What version of `ReactiveSearch API` are you using ?
commit hash: 8da3f178169fa60e78aa2c6c352ee78a667eee01


ubuntu 20

 go version
go version go1.17.7 linux/amd64

$ uname -a
Linux artcore-vb 5.4.0-100-generic #113-Ubuntu SMP Thu Feb 3 18:43:29 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux

i$ gcc -v
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-linux-gnu/9/lto-wrapper
OFFLOAD_TARGET_NAMES=nvptx-none:hsa
OFFLOAD_TARGET_DEFAULT=1
Target: x86_64-linux-gnu
Configured with: ../src/configure -v --with-pkgversion='Ubuntu 9.3.0-17ubuntu1~20.04' --with-bugurl=file:///usr/share/doc/gcc-9/README.Bugs --enable-languages=c,ada,c++,go,brig,d,fortran,objc,obj-c++,gm2 --prefix=/usr --with-gcc-major-version-only --program-suffix=-9 --program-prefix=x86_64-linux-gnu- --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --libdir=/usr/lib --enable-nls --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --with-default-libstdcxx-abi=new --enable-gnu-unique-object --disable-vtable-verify --enable-plugin --enable-default-pie --with-system-zlib --with-target-system-zlib=auto --enable-objc-gc=auto --enable-multiarch --disable-werror --with-arch-32=i686 --with-abi=m64 --with-multilib-list=m32,m64,mx32 --enable-multilib --with-tune=generic --enable-offload-targets=nvptx-none=/build/gcc-9-HskZEa/gcc-9-9.3.0/debian/tmp-nvptx/usr,hsa --without-cuda-driver --enable-checking=release --build=x86_64-linux-gnu --host=x86_64-linux-gnu --target=x86_64-linux-gnu
Thread model: posix
gcc version 9.3.0 (Ubuntu 9.3.0-17ubuntu1~20.04)

--- a/Dockerfile
+++ b/Dockerfile
@@ -1,4 +1,4 @@
-FROM golang:1.16.4-alpine3.13 as builder
+FROM golang:1.16.11 as builder
 
 # Default value
 # Run `--build-arg BILLING=true` to enable billing
@@ -27,7 +27,8 @@ ENV PLAN_REFRESH_INTERVAL="${PLAN_REFRESH_INTERVAL}"
 
 # Install tools required for project
 # Run `docker build --no-cache .` to update dependencies
-RUN apk add --no-cache build-base git
+RUN apt-get update
+RUN apt-get -y install build-essential git
 WORKDIR /reactivesearch
 
 # List project dependencies with go.mod and go.sum
@@ -42,11 +43,7 @@ COPY . .
 RUN make
 
 # Final stage: Create the running container
-FROM alpine:3.13 AS final
-
-# Get ca certs, for making api calls
-RUN apk add --no-cache ca-certificates
-
+FROM debian:bullseye AS final
 
 # Create env folder
 RUN mkdir /reactivesearch-data && touch /reactivesearch-data/.env && chmod 777 /reactivesearch-data/.env
@@ -56,4 +53,4 @@ COPY --from=builder /reactivesearch /reactivesearch
 WORKDIR /reactivesearch
 
 EXPOSE 8000
-ENTRYPOINT ["build/reactivesearch","--license-key-file", "--log", "stdout", "--plugins"]
+ENTRYPOINT ["build/reactivesearch", "--log", "stdout", "--plugins"]

